crates/fsrt/src/test.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{forge_project::ForgeProjectTrait, scan_directory, Args};
+use clap::Parser;
+use forge_analyzer::{definitions::PackageData, reporter::Vulnerability};
+use forge_loader::manifest::{ForgeManifest, FunctionMod};
+use std::fmt;
+use std::{
+    collections::{HashMap, HashSet},
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use swc_core::common::sync::Lrc;
+use swc_core::{
+    common::{FileName, SourceFile, SourceMap},
+    ecma::visit::FoldWith,
+};
+use tracing::debug;
+
+#[derive(Clone)]
+pub(crate) struct MockForgeProject<'a> {
+    pub files_name_to_source: HashMap<PathBuf, Arc<SourceFile>>,
+    pub test_manifest: ForgeManifest<'a>,
+    pub cm: Lrc<SourceMap>,
+}
+
+impl fmt::Debug for MockForgeProject<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Mock Forge Project {:?}", self.files_name_to_source)
+    }
+}
+
+#[allow(dead_code)]
+impl MockForgeProject<'_> {
+    pub fn files_from_string(string: String) -> Self {
```

Comment:
```suggestion
    pub fn files_from_string(string: &str) -> Self {
```

Saves a clone, since we end up having to create new strings anyway + it's more convenient.

Reply:
fixed. thank you :)